cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -616,6 +619,276 @@ async def _get_trustworthiness_score_async(
+            perplexity = [r["log"]["perplexity"] for r in prompt_response]
+            return self._batch_score(prompt, response, perplexity)
+
+        assert isinstance(prompt, str)
+        return self._score(
+            prompt,
+            prompt_response["response"],
+            perplexity=prompt_response["log"]["perplexity"],
+        )
+
+    def try_prompt(
+        self,
+        prompt: Sequence[str],
+    ) -> List[Optional[TLMResponse]]:
+        """
+        Gets response and trustworthiness score for any batch of prompts,
+        handling any failures (errors of timeouts) by returning None in place of the failures.
+
+        The list returned will have the same length as the input list, if there are any
+        failures (errors or timeout) processing some inputs, the list will contain None in place of the response.
+
+        This is the recommended way to get TLM responses and trustworthiness scores for big datasets of many prompts,
+        where some individual TLM responses within the dataset may fail. It ensures partial results are not lost.
+
+        Args:
+            prompt (Sequence[str]): list of multiple prompts for the TLM
+        Returns:
+            List[Optional[TLMResponse]]: list of [TLMResponse](#class-tlmresponse) objects containing the response and trustworthiness score.
+                The returned list will always have the same length as the input list.
+                In case of TLM failure on any prompt (due to timeouts or other errors),
+                the return list will contain None in place of the TLM response for that failed prompt.
+                Use this to obtain TLM results for as many prompts as possible,
+                but you might miss out on certain error messages.
+                If you prefer to be notified immediately about any errors or timeouts when running many prompts,
+                use the [`prompt()`](#method-prompt) method instead.
+        """
+        prompt_response = self.tlm_response.try_prompt(prompt)
+        prompt_succeeded_mask = [res is not None for res in prompt_response]
+
+        if not any(prompt_succeeded_mask):  # all prompts failed
+            return [None] * len(prompt)
+
+        # handle masking with numpy for easier indexing
+        prompt_succeeded = np.array(prompt)[prompt_succeeded_mask].tolist()
+        prompt_response_succeeded = np.array(prompt_response)[prompt_succeeded_mask]
+
+        response_succeeded = [r["response"] for r in prompt_response_succeeded]
+        perplexity_succeeded = [r["log"]["perplexity"] for r in prompt_response_succeeded]
+        score_response_succeeded = self._try_batch_score(
+            prompt_succeeded, response_succeeded, perplexity_succeeded
+        )
+
+        tlm_response = np.full(len(prompt), None)
+        tlm_response[prompt_succeeded_mask] = score_response_succeeded
+
+        return cast(List[Optional[TLMResponse]], tlm_response.tolist())
```

Comment:
could you just confirm that the returned values look exactly the same as from TLM.try_prompt() in batch-settings?

Reply:
Yup! It will be a `TLMResponse` dict with both response and score if it succeeds, and `None` if either step fails.